tensornetwork/tensornetwork.py

```diff
@@ -288,6 +289,7 @@ def __init__(self,
     self.axis1 = axis1
     self.node2 = node2
     self.axis2 = axis2
+    self._is_dangling = node2 is None
```

Comment:
I think there are some places that need to be updated now to check this flag instead of doing `node is None` or `node is not None` checks. Examples: l.808, l.835.

Reply:
Both of those are because of edge.get_edges() returning [edge.node1, edge.node2]. This is intended behavior.